doc/service_config.md

```diff
@@ -13,12 +13,20 @@ The service config is a JSON string of the following form:
 ```
 {
   // Load balancing policy name.
-  // Supported values are 'round_robin' and 'grpclb'.
-  // Optional; if unset, the default behavior is pick the first available
-  // backend.
-  // Note that if the resolver returns only balancer addresses and no
-  // backend addresses, gRPC will always use the 'grpclb' policy,
-  // regardless of what this field is set to.
+  // Currently, the only selectable client-side policy provided with gRPC
+  // is 'round_robin', but third parties may add their own policies.
+  // This field is optional; if unset, the default behavior is to pick
+  // the first available backend.
+  // If the policy name is set via the client API, that value overrides
+  // the value specified here.
+  //
+  // Note that if the resolver returns at least one balancer address (as
+  // opposed to backend addresses), gRPC will use grpclb (see
+  // https://github.com/grpc/grpc/blob/master/doc/load-balancing.md),
+  // regardless of what LB policy is requested either here or via the
+  // client API.  However, if the resolver returns backend addresses as
+  // well as balancer addresses, the client may fall back to the requested
+  // policy if it is unable to reach any of the grpclb load balancers.
```

Comment:
> No. People can get pick_first behavior by just not specifying any value here. We just don't want people to explicitly set this field to a value that is already the default value.

I think this is contradictory to your earlier comment:

> However, @a11r requested that we not tell people to explicitly specify the default value in their configs, because that makes it harder for us to change the default in the future.

Maybe you are conflating the users that just don't care and are willing to use whatever default we give them, and the users that explicitly want pick_first. Us changing the default in the future will serve right for the first kind, but will break the second, unless we allow the second to explicitly specify "pick_first".  I am not saying we should support the second kind, but instead asking whether we should.

> The core problem here is that there are two cases: (a) grpclb is in use and (b) client-side load balancing is in use. In case (a), you can indeed look at grpclb as the "primary" mechanism, and the client-side load balancing policy (the value specified here) is a "secondary" or "fallback" mechanism. But in case (b), the primary mechanism is the value specified here, so this is not a fallback.

Unfortunately I don't feel the same way. Maybe it's just me :)